### ISSUE TITLE
`http::basic_fields::equal_range` should return `boost::range::iterator_range`

### ISSUE THREAD

**Issue by LeonineKing1199** [boostorg/beast#1184]
In terms of improving Beast's semantic meaning, using a [`std::pair`](https://www.boost.org/doc/libs/1_67_0/libs/beast/doc/html/beast/ref/boost__beast__http__basic_fields/equal_range.html) as the return type for `basic_fields::equal_range` can lead to a hard-to-use API.

It's suggested that Beast instead return what's in Boost.Range for better interop with existing Boost libs.

**Comment by vinniefalco** [comment]
I rather not drag in Boost.Range just for one function in all of Beast. An alternative would be to roll a small range style class.

**Comment by klemens-morgenstern** [comment]
I wouldn't do it - this is the common pattern in all multiset/maps even today.  https://en.cppreference.com/w/cpp/container/multiset/equal_range

boost.range can adopt a pair into an iterator_range so I think consistency is better./

**Comment by vinniefalco** [comment]
I agree in principle, iterator pair is annoying. Boost.URL and Boost.Http-Proto both have `find_all`, e.g.:

All query parameters with a matching key:
```
subrange params::find_all( pct_string_view key );
```

All HTTP field values with a matching name:
```
subrange fields_view_base::find_all( string_view name );
```

`subrange` is a _ForwardRange_ with custom iterators that are convertible to the base container iterator.

If we do this for Beast, we should not change the current API but instead add a new one, e.g. `find_all`.

However, I don't see any particular pressing need for it.
